Source/NSStackView.m

```diff
@@ -0,0 +1,383 @@
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "AppKit/NSStackView.h"
+#import "GSFastEnumeration.h"
+
+@implementation NSStackView
+
+// Properties
+- (void) setDelegate: (id<NSStackViewDelegate>)delegate
+{
+  _delegate = delegate;
+}
+
+- (id<NSStackViewDelegate>) delegate
+{
+  return _delegate;
+}
+
+- (void) setOrientation: (NSUserInterfaceLayoutOrientation)o
+{
+  _orientation = o;
+}
+
+- (NSUserInterfaceLayoutOrientation) orientation
+{
+  return _orientation;
+}
+
+- (void) setAlignment: (NSLayoutAttribute)alignment
+{
+  _alignment = alignment;
+}
+
+- (NSLayoutAttribute) alignment
+{
+  return _alignment;
+}
+
+- (void) setEdgeInsets: (NSEdgeInsets)insets
+{
+  _edgeInsets = insets;
+}
+
+- (NSEdgeInsets) edgeInsets
+{
+  return _edgeInsets;
+}
+
+- (void) setDistribution: (NSStackViewDistribution)d
+{
+  _distribution = d;
+}
+
+- (NSStackViewDistribution) distribution
+{
+  return _distribution;
+}
+
+- (void) setSpacing: (CGFloat)f
+{
+  _spacing = f;
+}
+
+- (CGFloat) spacing
+{
+  return _spacing;
+}
+
+- (void) setDetachesHiddenViews: (BOOL)f
+{
+  _detachesHiddenViews = f;
+}
+
+- (BOOL) detachesHiddenViews
+{
+  return _detachesHiddenViews;
+}
+
+- (void) setArrangedSubviews: (NSArray *)arrangedSubviews
+{
+  ASSIGN(_arrangedSubviews, arrangedSubviews);
+}
+
+- (NSArray *) arrangedSubviews
+{
+  return _arrangedSubviews;
+}
+
+- (void) setDetachedViews: (NSArray *)detachedViews
+{
+  ASSIGN(_detachedViews, detachedViews);
+}
+
+- (NSArray *) detachedViews
+{
+  return _detachedViews;
+}
+
+// Instance methods
+// Manage views...
+- (instancetype) initWithViews: (NSArray *)views
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _arrangedSubviews = [[NSMutableArray alloc] initWithArray: views];
+      _detachedViews = [[NSMutableArray alloc] initWithCapacity: [views count]];
+      ASSIGNCOPY(_views, views);
+      _customSpacingMap = RETAIN([NSMapTable weakToWeakObjectsMapTable]);
+      _gravityMap = RETAIN([NSMapTable weakToWeakObjectsMapTable]);
+      _visiblePriorityMap = RETAIN([NSMapTable weakToWeakObjectsMapTable]);
+      _clippingMap = RETAIN([NSMapTable strongToStrongObjectsMapTable]);
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_arrangedSubviews);
+  RELEASE(_detachedViews);
+  RELEASE(_views);
+  RELEASE(_customSpacingMap);
+  RELEASE(_gravityMap);
+  RELEASE(_visiblePriorityMap);
+  RELEASE(_clippingMap);
+  _delegate = nil;
+  [super dealloc];
+}
+
++ (instancetype) stackViewWithViews: (NSArray *)views
+{
+  return [[self alloc] initWithViews: views];
```

Comment:
AUTORELEASE is missing here.

Reply:
You're correct.  Will fix this.   Welcome back.